test/shared/src/main/scala-2.x/zio/test/AssertionVariants.scala

```diff
@@ -23,11 +23,13 @@ trait AssertionVariants {
   /**
    * Makes a new assertion that requires a value equal the specified value.
    */
-  final def equalTo[A, B](expected: A)(implicit eql: Eql[A, B]): Assertion[B] =
-    Assertion.assertion("equalTo")(param(expected)) { actual =>
+  final def equalTo[A, B](expected: A)(implicit eql: Eql[A, B]): Assertion[B] = {
+    val assertion: Assertion[B] = Assertion.assertion("equalTo")(param(expected)) { actual =>
       (actual, expected) match {
         case (left: Array[_], right: Array[_]) => left.sameElements[Any](right)
         case (left, right)                     => left == right
       }
     }
+    assertion.withExpected(Some(expected))
```

Comment:
Do we want to add the expectation for every equality assertion or only those involving strings? This gets a little to the "when do we want to diff?" question.

Reply:
@adamgfraser see my latest commits.
It is still string only by default but you can customise it by passing a different `Diffing` to `equalTo`